src/components/Attachment/components/WaveProgressBar.tsx

```diff
@@ -55,10 +69,14 @@ export const WaveProgressBar = ({
       className='str-chat__wave-progress-bar__track'
       data-testid='wave-progress-bar-track'
       onClick={seek}
-      onMouseDown={handleMouseDown}
+      onMouseDown={handleDragStart}
       onMouseLeave={handleMouseUp}
       onMouseMove={handleMouseMove}
       onMouseUp={handleMouseUp}
+      onTouchEnd={handleMouseUp}
+      onTouchMove={handleTouchMove}
+      onTouchStart={handleDragStart}
```

Comment:
Don't want to send you down the rabbit hole, but if you want - check out pointer events instead of mouse and touch. Having a single interface might help to reduce code duplication.

Reply:
Thank you for the idea Matvei. I have made changes replacing those event listeners for pointer events listeners 👍